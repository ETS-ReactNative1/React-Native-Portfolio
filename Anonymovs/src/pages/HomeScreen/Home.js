import React, {useEffect, useState} from 'react';
import {
  StyleSheet,
  Text,
  View,
  Animated,
  FlatList,
  ScrollView,
  TouchableWithoutFeedback,
  ImageBackground,
  Image,
  TouchableOpacity,
} from 'react-native';
import Colors from '../../../assets/config/Colors';
import * as Animatable from 'react-native-animatable';
import {
  widthPercentageToDP as wp,
  heightPercentageToDP as hp,
} from 'react-native-responsive-screen';
import LinearGradient from 'react-native-linear-gradient';
import Icon from 'react-native-vector-icons/MaterialCommunityIcons';
import {useSelector, useDispatch} from 'react-redux';
import {
  actionDetail,
  actionGetAllMovies,
  actionReview,
  actionTopRated,
  actionUpcoming,
  actionVideoDetail,
} from '../../redux/action/actions';

export default function Home({navigation}) {
  const dispatch = useDispatch();
  const {movies} = useSelector(state => state.allReducers);
  const [selectCategory, setSelectCategory] = useState('1');
  useEffect(() => {
    dispatch(actionGetAllMovies());
  }, []);

  const categoryHandler = id => {
    setSelectCategory(id);
  };

  const category = [
    {id: '1', name: 'Now Playing'},
    {id: '2', name: 'Top Rated'},
    {id: '3', name: 'Upcoming'},
  ];

  const newSeasonScrollX = React.useRef(new Animated.Value(0)).current;
  const dotPosition = Animated.divide(newSeasonScrollX, wp(100));
  return (
    <View style={styles.container}>
      <Animatable.View animation="bounceInRight" style={styles.imageContainer}>
        <Animated.FlatList
          horizontal
          pagingEnabled
          snapToAlignment="center"
          snapToInterval={wp(100)}
          showsHorizontalScrollIndicator={false}
          scrollEventThrottle={16}
          decelerationRate={0}
          contentContainerStyle={{marginTop: hp(2)}}
          data={movies}
          keyExtractor={item => item.id}
          onScroll={Animated.event(
            [{nativeEvent: {contentOffset: {x: newSeasonScrollX}}}],
            {useNativeDriver: false},
          )}
          renderItem={({item, index}) => {
            return (
              <TouchableWithoutFeedback
                onPress={() => {
                  dispatch(actionDetail(item.id));
                  dispatch(actionVideoDetail(item.id));
                  dispatch(actionReview(item.id));
                  navigation.navigate('Detail');
                }}>
                <View style={styles.imageThumbnail}>
                  <ImageBackground
                    style={styles.img}
                    source={{
                      uri:
                        'https://image.tmdb.org/t/p/w500' + item.backdrop_path,
                    }}>
                    <LinearGradient
                      start={{x: 0, y: 0}}
                      end={{x: 0, y: 1}}
                      colors={['transparent', Colors.background]}
                      style={{
                        width: '100%',
                        height: hp(55),
                        alignItems: 'center',
                        justifyContent: 'flex-end',
                      }}>
                      <View style={styles.movieTitle}>
                        <Text style={styles.titleText}>{item.title}</Text>
                        <View
                          style={{flexDirection: 'row', alignItems: 'center'}}>
                          <Icon name="star" color="gold" size={20} />
                          <Text style={{color: Colors.white, marginLeft: 2}}>
                            {item.vote_average}
                          </Text>
                        </View>
                      </View>
                    </LinearGradient>
                  </ImageBackground>
                </View>
              </TouchableWithoutFeedback>
            );
          }}
        />
      </Animatable.View>
      <Animatable.View animation="bounceInRight" style={styles.detailContainer}>
        <View style={styles.dotWraper}>
          {movies.map((item, index) => {
            const opacity = dotPosition.interpolate({
              inputRange: [index - 1, index, index + 1],
              outputRange: [0.3, 1, 0.3],
              extrapolate: 'clamp',
            });
            const dotWidth = dotPosition.interpolate({
              inputRange: [index - 1, index, index + 1],
              outputRange: [6, 20, 6],
              extrapolate: 'clamp',
            });

            const dotColor = dotPosition.interpolate({
              inputRange: [index - 1, index, index + 1],
              outputRange: [Colors.unactive, Colors.active, Colors.unactive],
              extrapolate: 'clamp',
            });
            return (
              <Animated.View
                opacity={opacity}
                key={`dot-${index}`}
                style={{
                  borderRadius: 10,
                  marginHorizontal: 3,
                  width: dotWidth,
                  height: 6,
                  backgroundColor: dotColor,
                }}
              />
            );
          })}
        </View>
        <View style={styles.categoryItems}>
          {category.map(item => {
            return (
              <TouchableOpacity
                onPress={() => {
                  categoryHandler(item.id);
                  if (item.id === '1') {
                    dispatch(actionGetAllMovies());
                  } else if (item.id === '2') {
                    dispatch(actionTopRated());
                  } else {
                    dispatch(actionUpcoming());
                  }
                }}
                key={item.id}
                style={styles.itemWrap}>
                <Text
                  style={
                    selectCategory == item.id
                      ? styles.itemActive
                      : styles.itemUnactive
                  }>
                  {item.name}
                </Text>
              </TouchableOpacity>
            );
          })}
        </View>
      </Animatable.View>
      <View style={styles.posterWraper}>
        <FlatList
          data={movies}
          key={item => item.id}
          horizontal
          renderItem={({item}) => {
            return (
              <TouchableOpacity
                onPress={() => {
                  dispatch(actionDetail(item.id));
                  dispatch(actionVideoDetail(item.id));
                  dispatch(actionReview(item.id));
                  navigation.navigate('Detail');
                }}>
                <Image
                  style={styles.imgPoster}
                  source={{
                    uri: 'https://image.tmdb.org/t/p/w500' + item.poster_path,
                  }}
                />
              </TouchableOpacity>
            );
          }}
        />
      </View>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: Colors.background,
  },
  imageContainer: {
    flex: 2,
  },
  imageThumbnail: {
    backgroundColor: Colors.background,
    marginTop: -hp(2),
  },
  img: {
    width: wp(100),
    resizeMode: 'cover',
  },
  movieTitle: {
    width: wp(100),
    alignItems: 'center',
    justifyContent: 'flex-end',
    marginBottom: hp(5),
  },
  titleText: {
    color: Colors.white,
    fontSize: wp(6),
    textAlign: 'center',
  },
  detailContainer: {},
  dotWraper: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
  },
  categoryItems: {
    flexDirection: 'row',
    justifyContent: 'space-around',
    marginVertical: hp(3),
  },
  itemActive: {
    color: Colors.white,
    fontWeight: '700',
    fontSize: 16,
  },
  itemUnactive: {
    color: Colors.ash,
    fontWeight: '500',
  },
  posterWraper: {
    justifyContent: 'center',
    flex: 1,
  },
  imgPoster: {
    width: wp(32),
    height: hp(24),
    borderRadius: 6,
    marginRight: wp(2),
  },
});
