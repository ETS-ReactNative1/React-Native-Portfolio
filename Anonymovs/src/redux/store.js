import {createStore, applyMiddleware} from 'redux';
import createSagaMiddleware from 'redux-saga';
import sagas from './saga/sagas';
import combineReducers from './reducer/combine'

const sagaMiddleware = createSagaMiddleware();
const store = createStore(combineReducers,applyMiddleware(sagaMiddleware));

sagaMiddleware.run(sagas);

export default store;