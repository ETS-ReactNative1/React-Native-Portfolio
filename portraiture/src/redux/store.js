import {createStore, applyMiddleware} from 'redux';
import createSagaMiddleware from 'redux-saga';
import combineReducers from './reducers/combiner'
import sagas from './sagas/Sagas'

const sagaMiddleware = createSagaMiddleware();
const store = createStore(combineReducers,applyMiddleware(sagaMiddleware));

sagaMiddleware.run(sagas);

export default store;