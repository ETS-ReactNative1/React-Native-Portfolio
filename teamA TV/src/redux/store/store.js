import ReduxThunk from 'redux-thunk';
import {createStore, applyMiddleware} from 'redux';
import rootReducer from '../reducer/combineReducer'


export let store = createStore(rootReducer,applyMiddleware(ReduxThunk))
