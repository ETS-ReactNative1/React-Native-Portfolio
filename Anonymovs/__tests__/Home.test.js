import 'react-native';
import React from 'react';
import {FlatList} from 'react-native';
import Home from '../src/pages/HomeScreen/Home';
import renderer from 'react-test-renderer';
import {Provider} from 'react-redux';
import store from '../src/redux/store';
import {render,waitForElement, fireEvent} from '@testing-library/react-native';
import axios from 'axios'
jest.useFakeTimers();

test('Home SnapShoot', () => {
  const snap = renderer
    .create(
      <Provider store={store}>
        <Home />
      </Provider>,
    )
    .toJSON();

  expect(snap).toMatchSnapshot();
});
it(`renders correctly`, () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <Home />
      </Provider>,
    )
    .toJSON();
  expect(tree.children.length).toBe(3);
});

const mockData = [
  {
    id: 'id-1',
    title: 'Title-1',
  },
  {
    id: 'id-2',
    title: 'Title-2',
  },
  {
    id: 'id-3',
    title: 'Title-3',
  },
];

describe('Testing FlatList', () => {
  test('Error component should be render when error is true', () => {
    const componentTree = render(
      <Provider store={store}>
        <Home data={mockData} />
      </Provider>,
    );

    expect(componentTree.UNSAFE_getAllByType(FlatList).length).toBe(2);
  });
});