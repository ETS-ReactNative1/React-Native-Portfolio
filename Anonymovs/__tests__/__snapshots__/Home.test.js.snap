// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home SnapShoot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#191926",
      "flex": 1,
    }
  }
>
  <View
    forwardedRef={[Function]}
    style={
      Object {
        "flex": 2,
        "opacity": 0,
        "transform": Array [
          Object {
            "translateX": 600,
          },
        ],
      }
    }
  >
    <RCTScrollView
      contentContainerStyle={
        Object {
          "marginTop": 26.5,
        }
      }
      data={Array []}
      decelerationRate={0}
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={true}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      pagingEnabled={true}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={16}
      showsHorizontalScrollIndicator={false}
      snapToAlignment="center"
      snapToInterval={750}
      stickyHeaderIndices={Array []}
      style={Object {}}
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <View />
    </RCTScrollView>
  </View>
  <View
    forwardedRef={[Function]}
    style={
      Object {
        "opacity": 0,
        "transform": Array [
          Object {
            "translateX": 600,
          },
        ],
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    />
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-around",
          "marginVertical": 40,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#FFFFFF",
              "fontSize": 16,
              "fontWeight": "700",
            }
          }
        >
          Now Playing
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#999999",
              "fontWeight": "500",
            }
          }
        >
          Top Rated
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#999999",
              "fontWeight": "500",
            }
          }
        >
          Upcoming
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <RCTScrollView
      data={Array []}
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={true}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <View />
    </RCTScrollView>
  </View>
</View>
`;
